provider/openstack/firewaller.go

```diff
@@ -950,13 +957,16 @@ func (c *neutronFirewaller) closePortsInGroup(ctx context.ProviderCallContext, n
 	}
 	neutronClient := c.environ.neutron()
 	// TODO: Hey look ma, it's quadratic
+	// TODO: (stickupkid): Put into a map and delete linearly.
 	for _, rule := range rules {
 		for _, p := range group.Rules {
 			if !secGroupMatchesIngressRule(p, rule) {
 				continue
 			}
-			err := neutronClient.DeleteSecurityGroupRuleV2(p.Id)
-			if err != nil {
+			if err := neutronClient.DeleteSecurityGroupRuleV2(p.Id); err != nil {
+				if gooseerrors.IsNotFound(err) {
+					continue
```

Comment:
continue or break?

Reply:
`continue`, as we want to continue attempting to clear the rest of the security groups.

Reply:
No, you're right, it should be break.